src/eth/TransactionManager.js

```diff
@@ -210,18 +210,18 @@ class Tracker {
     return txs[0];
   }
 
-  clearExpiredListeners() {
-    each(this._transactions, (val, key) => {
-      val.forEach(tx => {
-        // TODO: check Tx age for 5 minutes or older:
-        // const age = (new Date() - new Date(tx._timeStampMined)) / 1000;
-        if (tx.isFinalized() || tx.isError()) {
-          log(
-            `deleting key ${key}, contract: ${tx.metadata.contract}, method: ${
-              tx.metadata.method
-            }`
-          );
-          delete this._transactions[key];
+  clearExpiredTransactions() {
+    each(this._transactions, (txList, key) => {
+      txList.forEach(tx => {
+        const txAge =
+          new Date(tx._timeStampMined).getMinutes() - new Date().getMinutes();
+        if ((tx.isFinalized() || tx.isError()) && txAge > 5) {
```

Comment:
So there's a nuance here that I missed the first time around -- `isFinalized` will never be true for a given tx unless `confirm` has been called on it. But there's no way of knowing from here whether the user intends to call `confirm` or not. So the check we should do here is probably just `isMined() || isError` plus the timeout check, assuming that if the user does want to call `confirm` and/or register a `confirm` listener, that that will all happen before the timeout.

On the other hand--maybe it's better to remove this restriction, "`isFinalized` will never be true for a given tx unless `confirm` has been called on it". We could, for example, just-in-time compare the current block number as reported by the web3 service to the transaction's block number, when `isFinalized` is called, to determine its truth regardless of whether `confirm` was called. Want to try that?

Reply:
Yeah definitely, would this change be to the `isFinalized` method of the transaction lifecycle class? Something like this:
` isFinalized(bn = null) {
    return bn ? this._web3Service.blockNumber() === bn.blockNumber : this._state.inState(finalized);
  }`

Reply:
I think we can store the necessary values in the transaction object so we don't need to pass an argument. The information we need is already in TransactionObject; it just needs to be pushed down into TransactionLifecycle so isFinalized can use it.

Need 3 values:
- the web3 service
  - add it to the constructor of TransactionLifecycle
- the block number when mined
  - this can be stored after mining, just like we do for `this._timestampMined` 
- the number of blocks to wait for confirmation
  - if `confirm` is called, we can just store the argument
  - if `confirm` isn't called, we can use `web3Service.confirmedBlockCount()`

then isFinalized would be something like this:
```
  return this._state.inState(finalized) || this._blockNumberWhenMined + this._confirmedBlockCount <= this._web3Service.blockNumber()
```

Reply:
That makes sense. One question, it looks like `confirmedBlockCount()` is already available to the TransactionLifecycle class. Do I need to update it again somewhere?